src/serac/numerics/functional/detail/triangle_H1.inl

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2019-2023, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file triangle_H1.inl
+ *
+ * @brief Specialization of finite_element for H1 on triangle geometry
+ */
+
+// this specialization defines shape functions (and their gradients) that
+// interpolate at Gauss-Lobatto nodes for the appropriate polynomial order
+//
+// note: mfem assumes the parent element domain is the convex hull of {{0,0}, {1,0}, {0,1}}
+// for additional information on the finite_element concept requirements, see finite_element.hpp
+/// @cond

```

Comment:
Are these large exclusions necessary/wanted?

Reply:
the `detail/*.inl` files all implement different template specializations of the same class, so `@cond` is used here to avoid writing the same doxygen descriptions on each specialization.

There is a more detailed docstring on the "base class":
https://github.com/LLNL/serac/blob/71e4dadb5365ccfef7a30f5f342098948c0a30da/src/serac/numerics/functional/finite_element.hpp#L324-L348